src/vs/workbench/common/editor/untitledTextEditorModel.ts

```diff
@@ -242,7 +242,8 @@ export class UntitledTextEditorModel extends BaseTextEditorModel implements IUnt
 	}
 
 	private updateNameFromFirstLine(): void {
-		if (this.hasAssociatedFilePath) {
+		// {{SQL CARBON EDIT}} - For Query Editor, must not change name even without file path.
+		if (!this.hasAssociatedFilePath && this.hasAssociatedFilePath) {
```

Comment:
I'm not understanding this change - don't we want to return here? The statement you have is always going to evaluate to false so it'll always use the first line text...

Reply:
woops, I meant either.